src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2212,7 +2285,11 @@ public static bool TrueForAll<T>(T[] array, Predicate<T> match)
 
         public unsafe Object GetValue(int index)
         {
+#if REAL_MULTIDIM_ARRAYS
+            if (!this.EETypePtr.IsSzArray)
```

Comment:
We can actually `bool IsSzArray` property directly on `System.Array` - that just checks the BaseSize without checkin IsArray - because of we know that it is array already.